src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/AbstractSourceFilePropertyProvider.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Properties
+{
+    /// <summary>
+    /// A provider for properties that are stored in the source code of the project.
+    /// </summary>
+    internal abstract class AbstractSourceFilePropertyProvider : IProjectPropertiesProvider
```

Comment:
Nit: in most places we are using "Properties" in the name PropertiesProvider (including the derived class SourceFilePropertiesProvider)